presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkTaskSourceRdd.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import org.apache.spark.Dependency;
+import org.apache.spark.InterruptibleIterator;
+import org.apache.spark.Partition;
+import org.apache.spark.SparkContext;
+import org.apache.spark.TaskContext;
+import org.apache.spark.rdd.ParallelCollectionPartition;
+import org.apache.spark.rdd.RDD;
+import scala.collection.Iterator;
+import scala.reflect.ClassTag;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static scala.collection.JavaConversions.asScalaBuffer;
+
+public class PrestoSparkTaskSourceRdd
```

Comment:
For now Looks like it's a wrapper over `ParallelCollectionPartition` . I assume it will have non-trivial functionality when colocated join is supported :)

Reply:
In fact I need to take a closer look into this. Looks like it's more than just a wrapper :) (especially this `getInputIterators` method)

Reply:
> For now Looks like it's a wrapper over ParallelCollectionPartition . I assume it will have non-trivial functionality when colocated join is supported :)

Yeah. It provides us with a flexibility to manually assign `TaskSource` to spark partitions.  `ParallelRdd` (the result of `parallelize`) just does round robin.